lib/private/httphelper.php

```
@@ -215,6 +215,7 @@ public function post($url, array $fields) {
 		curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
 		curl_setopt($ch, CURLOPT_POST, count($fields));
 		curl_setopt($ch, CURLOPT_POSTFIELDS, $fieldsString);
+		curl_setopt($ch, CURLOPT_TIMEOUT, 10);
```

`CURLOPT_TIMEOUT` seems like the wrong option to me. According to the documentation from http://curl.haxx.se/libcurl/c/CURLOPT_TIMEOUT.html:
> Pass a long as parameter containing timeout - the maximum time in seconds that you allow the libcurl transfer operation to take. 

So this will fail hard in a lot of cases where the response takes more than 10 seconds. There are legitimate cases where this happens.

`CURLOPT_CONNECTTIMEOUT` looks way better to me as can be seen at http://curl.haxx.se/libcurl/c/CURLOPT_CONNECTTIMEOUT.html:

> Pass a long. It should contain the maximum time in seconds that you allow the connection phase to the server to take. This only limits the connection phase, it has no impact once it has connected. Set to zero to switch to the default built-in connection timeout - 300 seconds. See also the CURLOPT_TIMEOUT option.